### ISSUE TITLE
error

### ISSUE THREAD

**Issue by hone5t** [johnfmorton/Pic-Puller-for-Instagram#1]
i keep getting this error 
the setup is fine and the authorization is done any idea what causing this error
No data returned from Instagram API. Check http://api-status.com/6404/174981/Instagram-API

**Comment by johnfmorton** [comment]
On June 1, Instagram killed API access for almost all Instagram apps. Basically, my guess is your Instagram app that you created with Pic Puller was automatically put into sandbox mode by Instagram. What this does is that it severely limits the data you can retrieve from Instagram. 

The only thing sandboxed apps have access to are the 20 most recent pieces of media uploaded to the authorized user's account.

Does this sound like what's happened in your situation?

**Comment by hone5t** [comment]
thanks for your reply its seems my application is actually in sandbox mode now but i still dont get why i get the error its suppose to give me the last 20 uploaded image right???

**Comment by johnfmorton** [comment]
Which function are you using? I would expect the "media_recent" function to work for the last 20 photos. Can you share a screenshot of how the error is being presented? Maybe that will help me understand what you're encountering.

**Comment by hone5t** [comment]
i am using media_recent 
![screen shot 2016-06-13 at 5 17 53 pm](https://cloud.githubusercontent.com/assets/960415/16025111/0c2e4e46-318b-11e6-910d-b0f6fd307789.png)

**Comment by johnfmorton** [comment]
One other question, how are you authorizing the user? The reason I ask is that as Pic Puller matured I added the ability for it to pull pictures from a user other than the one to authorize the app. This was useful when you as a developer did not have the username and password for your client but simply needed to import their images. In that scenario you could specify your clients Instagram userid in the loop and pull in their images.

In sandbox mode you only have access to the images for the authorized user. Trying to access a third-party's images would cause an error because you do not have access to "public" media via the application anymore. Could this be the issue you're running into?

**Comment by hone5t** [comment]
i am accessing the image for the same user and its work like for 10 min or so then i get this error :(

**Comment by johnfmorton** [comment]
Hmm. So it works and then you're getting this error? And, when you get the error, you're not retrieving the last successful JSON data from your cache? Interesting. This makes me think you're running into the caching limit of the sandboxed app. I don't know what that is specifically, but Instagram says it's much lower than an approved app. 

Are you you caching your templates? 

Also, what is your pic puller tag set to do with regard to caching?

**Comment by hone5t** [comment]
below is my tag 
 {exp:ig_picpuller:media_recent user_id="75143" use_stale_cache="yes" limit="8"}
and i know the limit is lower for sandbox but when i use just link with token using ajax call i can get the data alright

**Comment by johnfmorton** [comment]
With the tag you're using, I would expect the _cached data_ to be used when you hit the API limit. I wonder if the permissions on your cache directory might be preventing the cache from being written to disk. 

On my end I have the permissions of the cache directory set to 777. 

My guess at the location of this on your end: 
`/system/expressionengine/cache`

When you go into that directory, do you see any directories that look something like "ig_picpuller_mysitename" or just "ig_picpuller"? That's where PP tries to write the cache for each successful request from Instagram.

**Comment by goodmixer** [comment]
Was having the same issue with same error. Turns out the max_id='{segment_3}' portion of the following tag was causing the error 
{exp:ig_picpuller:media_recent user_id='1' limit='10' max_id='{segment_3}' use_stale_cache='yes'}

Couldn't see any data written to the ig_picpuller folder in my cache either. Permissions set to 777 and had to create folder manually.

**Comment by daniellelecomte** [comment]
Question - with the limit of 20 most recent, does that mean that older images will no longer display?

**Comment by johnfmorton** [comment]
Yes, since your app will be in [Sandbox Mode](https://www.instagram.com/developer/sandbox/) the universe that your Instagram app can see into only includes your most recent 20 images. As soon as an image is the 21st newest image in your Instagram stream, as far as your app is concerned, it does not exist.
